contracts/src/v0.8/dev/keeper2_0/KeeperRegistry2_0.sol

```diff
@@ -40,144 +42,173 @@ contract KeeperRegistry2_0 is
-   * @param registryGasOverhead the gas overhead used by registry in performUpkeep
    * @param link address of the LINK Token
-   * @param linkEthFeed address of the LINK/ETH price feed
+   * @param linkNativeFeed address of the LINK/Native price feed
    * @param fastGasFeed address of the Fast Gas price feed
-   * @param config registry config settings
+   * @param onChainConfig registry on chain config settings
    */
   constructor(
     PaymentModel paymentModel,
-    uint256 registryGasOverhead,
     address link,
-    address linkEthFeed,
+    address linkNativeFeed,
     address fastGasFeed,
     address keeperRegistryLogic,
-    Config memory config
-  ) KeeperRegistryBase2_0(paymentModel, registryGasOverhead, link, linkEthFeed, fastGasFeed) {
+    OnChainConfig memory onChainConfig
+  ) KeeperRegistryBase2_0(paymentModel, link, linkNativeFeed, fastGasFeed) {
     KEEPER_REGISTRY_LOGIC = keeperRegistryLogic;
-    setConfig(config);
+    setOnChainConfig(onChainConfig);
   }
 
+  ////////
   // ACTIONS
+  ////////
 
   /**
-   * @notice adds a new upkeep
-   * @param target address to perform upkeep on
-   * @param gasLimit amount of gas to provide the target contract when
-   * performing upkeep
-   * @param admin address to cancel upkeep and withdraw remaining funds
-   * @param checkData data passed to the contract when checking for upkeep
+   * @dev This struct is used to maintain run time information about an upkeep in transmit function
+   * @member upkeep the upkeep struct
+   * @member earlyChecksPassed whether the upkeep passed early checks before perform
+   * @member paymentParams the paymentParams for this upkeep
+   * @member performSuccess whether the perform was successful
+   * @member gasUsed gasUsed by this upkeep in perform
    */
-  function registerUpkeep(
-    address target,
-    uint32 gasLimit,
-    address admin,
-    bytes calldata checkData
-  ) external override returns (uint256 id) {
-    // Executed through logic contract
-    _fallback();
+  struct UpkeepTransmitInfo {
+    Upkeep upkeep;
+    bool earlyChecksPassed;
+    PerformPaymentParams paymentParams;
+    bool performSuccess;
+    uint256 gasUsed;
   }
 
   /**
-   * @notice simulated by keepers via eth_call to see if the upkeep needs to be
-   * performed. If upkeep is needed, the call then simulates performUpkeep
-   * to make sure it succeeds. Finally, it returns the success status along with
-   * payment information and the perform data payload.
-   * @param id identifier of the upkeep to check
-   * @param from the address to simulate performing the upkeep from
+   * @inheritdoc OCR2Abstract
    */
-  function checkUpkeep(uint256 id, address from)
-    external
-    override
-    cannotExecute
-    returns (
-      bytes memory performData,
-      uint256 maxLinkPayment,
-      uint256 gasLimit,
-      uint256 adjustedGasWei,
-      uint256 linkEth
-    )
-  {
-    // Executed through logic contract
-    _fallback();
+  function transmit(
```

Comment:
do we have benchmarks for realistic DON sizes, eg 16 nodes

Reply:
initial benchmark, f=1, overhead = 145K. Need to calculate for f=5